core/class.go

```diff
@@ -46,8 +57,270 @@ func (c *Cairo0Class) Version() uint64 {
+}
+
+func makeDeprecatedVMClass(class *Cairo0Class) (*starknet.Cairo0Definition, error) {
+	adaptEntryPoint := func(ep EntryPoint) starknet.EntryPoint {
+		return starknet.EntryPoint{
+			Selector: ep.Selector,
+			Offset:   ep.Offset,
+		}
+	}
+
+	constructors := utils.Map(utils.NonNilSlice(class.Constructors), adaptEntryPoint)
+	external := utils.Map(utils.NonNilSlice(class.Externals), adaptEntryPoint)
+	handlers := utils.Map(utils.NonNilSlice(class.L1Handlers), adaptEntryPoint)
+
+	decompressedProgram, err := utils.Gzip64Decode(class.Program)
+	if err != nil {
+		return nil, err
+	}
+
+	return &starknet.Cairo0Definition{
+		Program: decompressedProgram,
+		Abi:     class.Abi,
+		EntryPoints: starknet.EntryPoints{
+			Constructor: constructors,
+			External:    external,
+			L1Handler:   handlers,
+		},
+	}, nil
+}
+
+// applyReplacer recursively applies the replacer function to the JSON data
+func applyReplacer(data any, replacer func(string, any) any) any {
+	switch v := data.(type) {
+	case map[string]any:
+		for key, val := range v {
+			v[key] = applyReplacer(replacer(key, val), replacer)
+			if v[key] == nil && key != debugInfo {
+				delete(v, key)
+			}
+		}
+
+		return v
+	case []any:
+		for i, val := range v {
+			v[i] = applyReplacer(replacer("", val), replacer)
+		}
+		return v
+	case *orderedmap.OrderedMap[string, any]:
+		for pair := v.Oldest(); pair != nil; pair = pair.Next() {
+			val := applyReplacer(replacer(pair.Key, pair.Value), replacer)
+			if val == nil {
+				v.Delete(pair.Key)
+			} else {
+				v.Set(pair.Key, val)
+			}
+		}
+		return v
+	default:
+		return replacer("", v)
+	}
+}
+
+// nullSkipReplacer is a custom JSON replacer that handles specific keys and null values
+func nullSkipReplacer(key string, value any) any {
+	switch key {
+	case "attributes", "accessible_scopes", "flow_tracking_data":
+		if arr, ok := value.([]any); ok && len(arr) == 0 {
+			return nil
+		}
+	case debugInfo:
+		return nil
+	}
+
+	return value
+}
+
+// identifiersNullSkipReplacer is same as nullSkipReplacer with an addition condition
+// on the `cairo_type` field.
+// Used only in the `identifiers` field.
+func identifiersNullSkipReplacer(key string, value any) any {
+	switch key {
+	case "cairo_type":
+		if str, ok := value.(string); ok {
+			return strings.ReplaceAll(str, ": ", " : ")
+		}
+	case "attributes", "accessible_scopes", "flow_tracking_data":
+		if arr, ok := value.([]any); ok && len(arr) == 0 {
+			return nil
+		}
+	case debugInfo:
+		return nil
+	}
+
+	return value
+}
+
+// stringify converts a Go value to a JSON string, using a custom replacer function
+func stringify(value any, replacer func(string, any) any) (string, error) {
+	// Marshal the value to JSON
+	jsonBytes, err := json.Marshal(value)
+	if err != nil {
+		return "", err
+	}
+
+	// Determine the type of the JSON data
+	var jsonData any
+	if err := json.Unmarshal(jsonBytes, &jsonData); err != nil {
+		return "", err
+	}
+
```

Comment:
isn't `jsonData` is the same as `value` at this point ?

Reply:
I changed `value` from `any` to `json.RawMessage`